presto-main/src/main/java/com/facebook/presto/execution/CreateFunctionTask.java

```diff
@@ -82,13 +85,21 @@ public String explain(CreateFunction statement, List<Expression> parameters)
             throw new PrestoException(NOT_SUPPORTED, "Invoking a dynamically registered function in SQL function body is not supported");
         }
 
-        metadata.getFunctionManager().createFunction(createSqlInvokedFunction(statement, metadata, analysis), statement.isReplace());
+        SqlInvokedFunction function = createSqlInvokedFunction(statement, metadata, analysis);
+        if (statement.isTemporary()) {
+            stateMachine.addSessionFunction(new BasicSignature(function.getSignature()), function);
```

Comment:
isReplace check here?

Reply:
Analyzer will check for REPLACE, and throw if it is used with TEMPORARY.